x-pack/plugins/observability/public/components/shared/exploratory_view/series_builder/series_builder.tsx

```diff
@@ -0,0 +1,162 @@
+
+  const [dataType, setDataType] = useState<AppDataType | null>(null);
+  const [reportType, setReportType] = useState<ReportViewTypeId | null>(null);
+  const [serviceName, setServiceName] = useState<string | null>(null);
+
+  const columns = [
+    {
+      name: 'DataType',
+      field: 'dataType',
+      width: '20%',
+      render: (val: string) => <DataTypesCol selectedDataType={dataType} onChange={setDataType} />,
+    },
+    {
+      name: 'Report',
+      field: 'defaultFilters',
+      width: '30%',
+      render: (val: string) => (
+        <ReportTypesCol
+          selectedReportType={reportType}
+          reportTypes={dataType ? ReportTypes[dataType] : []}
+          onChange={setReportType}
+        />
+      ),
+    },
+    {
+      name: 'Definition',
+      field: 'defaultFilters',
+      width: '30%',
+      render: (val: string) =>
+        reportType ? (
+          <ReportDefinitionCol
+            dataType={dataType}
+            selectedReportDefinitions={null}
+            reportType={reportType}
+            onChange={setServiceName}
+          />
+        ) : null,
+    },
+    {
+      name: 'Filters',
+      field: 'filters',
+      width: '25%',
+      render: (val: string[]) => (reportType ? <ReportFilters /> : null),
+    },
+    {
+      name: 'Breakdowns',
+      width: '25%',
+      field: 'id',
+      render: (val: string[]) => (reportType ? <ReportBreakdowns /> : null),
+    },
+  ];
+
+  const { setSeries, allSeriesIds } = useUrlStorage();
+
+  const addSeries = () => {
+    if (reportType) {
+      const newSeriesId = `${serviceName!}-${reportType}`;
+
+      const newSeries = {
+        [REPORT_TYPE]: reportType,
+        serviceName,
+        time: { from: 'now-30m', to: 'now' },
+      } as SeriesUrl;
+      setSeries(newSeriesId, newSeries);
+
+      // reset state
+      setDataType(null);
+      setReportType(null);
+      setServiceName(null);
+      setIsFlyoutVisible(false);
+    }
+  };
+
+  const items = [{ dataTypes: ['APM'] }];
+
+  let flyout;
+
+  if (isFlyoutVisible) {
+    flyout = (
+      <BottomFlyout aria-labelledby="flyoutTitle">
+        <EuiBasicTable
+          items={items}
+          columns={columns}
+          cellProps={{ style: { borderRight: '1px solid #d3dae6' } }}
+        />
+        <EuiSpacer />
+        <EuiFlexGroup justifyContent="flexEnd">
+          <EuiFlexItem grow={false}>
+            <EuiButton iconType="plus" color="primary" disabled={!serviceName} onClick={addSeries}>
+              Add
+            </EuiButton>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton iconType="cross" color="danger">
+              Cancel
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </BottomFlyout>
+    );
+  }
+
+  return (
+    <div>
+      <EuiButton
+        iconType={isFlyoutVisible ? 'arrowDown' : 'arrowRight'}
+        color="secondary"
```

Comment:
I would recommend using the primary color. I think this is the most important button on this screen when there is no series yet.

```suggestion
        color="primary"
```